app/geant-exporter/geant-exporter.cc

```diff
@@ -334,55 +276,133 @@ void store_geometry(TFile*                       root_file,
 
-    // Recursive loop over all logical volumes, starting from the world_volume
-    // Populate volume information and map volumes with materials
-    loop_volumes(geometry, *world_volume.GetLogicalVolume());
+    CELER_LOG(info) << "Added " << materials.size() << " materials";
+    return materials;
+}
 
-    tree_materials.Fill();
-    int err_code = root_file->Write();
-    CELER_ENSURE(err_code >= 0);
+//---------------------------------------------------------------------------//
+/*!
+ * Return a populated \c ImportProcess vector.
+ */
+std::vector<ImportProcess> store_processes()
+{
+    CELER_LOG(status) << "Exporting physics processes";
+
+    std::vector<ImportProcess> processes;
+    ImportProcessConverter     process_writer(TableSelection::minimal);
+
+    G4ParticleTable::G4PTblDicIterator& particle_iterator
+        = *(G4ParticleTable::GetParticleTable()->GetIterator());
+    particle_iterator.reset();
+
+    while (particle_iterator())
+    {
+        const G4ParticleDefinition& g4_particle_def
+            = *(particle_iterator.value());
+
+        celeritas::PDGNumber pdg(g4_particle_def.GetPDGEncoding());
+        if (!pdg)
+        {
+            // Skip "dummy" particles: generic ion and geantino
+            continue;
+        }
+
+        // XXX To reduce ROOT file data size in repo, only export processes for
+        // electron/positron/gamma for now. Extend this later.
+        if (!(pdg == celer_pdg::electron() || pdg == celer_pdg::positron()
+              || pdg == celer_pdg::gamma()))
+        {
+            // Not e-, e+, or gamma
+            continue;
+        }
+
+        const G4ProcessVector& process_list
+            = *g4_particle_def.GetProcessManager()->GetProcessList();
+
+        for (auto j : celeritas::range(process_list.size()))
+        {
+            if (dynamic_cast<const G4Transportation*>(process_list[j]))
+            {
+                // Skip transportation process
+                continue;
+            }
+
+            processes.push_back(
+                process_writer(g4_particle_def, *process_list[j]));
+        }
+    }
+    // Remove empty processes returned by operator()
+    process_writer.remove_empty(&processes);
+    CELER_LOG(info) << "Added " << processes.size() << " processes";
+    return processes;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return a populated \c ImportVolume vector.
+ */
+std::vector<ImportVolume> store_volumes(const G4VPhysicalVolume* world_volume)
+{
+    CELER_LOG(status) << "Exporting volumes";
+
+    std::vector<ImportVolume>            volumes;
+    std::map<unsigned int, ImportVolume> volids_volumes;
+
+    // Recursive loop over all logical volumes to populate map<volid, volume>
+    loop_volumes(volids_volumes, *world_volume->GetLogicalVolume());
+
+    // Populate vector<ImportVolume>
+    for (const auto& key : volids_volumes)
+    {
+        volumes.push_back(key.second);
```

Comment:
One final change: if someone unwittingly replaces `map` with `unordered_map`, or if the logical volumes are discontiguous/offset for some reason, the loop will proceed as normal and the IDs  in the export file will be non-obviously mixed up.
```suggestion
    for (const auto& key : volids_volumes)
    {
        CELER_ASSERT(key.first == volumes.size());
        volumes.push_back(key.second);
```